src/Clash/Cores/Ethernet/Mdio.hs

```diff
@@ -0,0 +1,334 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RecordWildCards #-}
+
+{- |
+  Copyright  :  (C) 2024, QBayLogic B.V.
+  License    :  BSD2 (see the file LICENSE)
+  Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
+
+Provides an MDIO bus controller as specified in IEEE 802.3 Clause 22.
+-}
+module Clash.Cores.Ethernet.Mdio (
+  MdioOutput (..),
+  MdioRequest (..),
+  MdioResponse (..),
+  mdioController,
+) where
+
+import Control.DeepSeq (NFData)
+
+import Clash.Prelude
+
+-- | MDIO bus request.
+data MdioRequest
+  = MdioRead
+      { mdioSuppressPreamble :: Bool
```

Comment:
Isn't this really fixed for a specific Phy? If that is this case I would say make this an argument when instantiating rather then inside each request.

Reply:
I just realized you could have multiple different PHYs on the bus where some support suppression and some do not.